pkg/vip/address.go

```
@@ -710,10 +710,28 @@ func (configurator *network) SetMask(mask string) error {
 	if err != nil {
 		return err
 	}
+
 	size := 32
-	if IsIPv6(configurator.address.IP.String()) {
+	family := "IPv4"
+
+	if IsIPv6(configurator.IP()) {
 		size = 128
+		family = "IPv6"
+	}
+
+	if m > size {
+		log.Warn("provided mask is greater than the highest mask value for the IP family - will use max possible value", "family", family, "mask", m, "max", size)
+		m = size
```

I would suggest that it should error.
Better fail fast then assuming config.

---

Done :)

---

Thx :)